### PR TITLE
Issue 471: Fix panic when removing a route that doesn't exist in w.routes.

### PR DESCRIPTION
**Pull request by jeffreydwalter**
This PR fixes the remaining panic in `RemoveRoute()`.

### CODE REVIEW

File: web_service.go [hunk 1]
```diff
@@ -176,22 +176,20 @@ func (w *WebService) Route(builder *RouteBuilder) *WebService {
...
-		current++
-	}
-	w.routes = newRoutes
-	return nil
+    if !w.dynamicRoutes {
+        return errors.New("dynamic routes are not enabled.")
+    }
+    w.routesLock.Lock()
+    defer w.routesLock.Unlock()
+    newRoutes := make([]Route, 0)
```
Comment (emicklei) [inline web_service.go:184]: now that the length is not of interest, we can write []Route{}.

File: web_service.go [hunk 2]
```diff
@@ -176,22 +176,20 @@ func (w *WebService) Route(builder *RouteBuilder) *WebService {
...
+    for ix := range w.routes {
```
Comment (emicklei) [inline web_service.go:185]: now that the index is not needed, we can write    for _ , each := range w.routes {   if each.Method == method.... }
so using `each` for `w.routes[ix]`

**Comment by jeffreydwalter** [comment]
@emicklei I made the changes you suggested. Would like to get this PR merged so I can vendor it back into my codebase. Please let me know if there's anything else I can do to facilitate. Thanks!

**Comment by jeffreydwalter** [comment]
@emicklei thinking about this further, what do you think about setting the capacity of `newRoutes`, like this: `newRoutes := make([]Route, 0, len(w.routes))`? This would potentially allocate an extra item in the slice in the case the route is found and removed, but would cut down on slice allocations. If you think that is better, I'm happy to make another PR.

**Comment by jeffreydwalter** [comment]
@emicklei what do you say in regard to my last comment? ☝🏻

**Comment by emicklei** [comment]
@jeffreydwalter , removing a Route is not on the hot-path so any optimisations here are not significant.

**Pull request merged**